src/Framework/Http/ConnectServer/Client/ConnectClient.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace Give\Framework\Http\ConnectServer\Client;
+
+use Give\Framework\Http\ConnectServer\Client\Exceptions\RequestException;
+use WP_Error;
+
+/**
+ * Class ConnectClient
+ *
+ * @since   2.8.0
+ */
+class ConnectClient
+{
+    /**
+     * Api Url
+     *
+     * @since 2.8.0
+     *
+     * @var string
+     */
+    public $apiUrl = 'https://connect.givewp.com';
+
+    /**
+     * @unreleased
+     */
+    public function __construct($giveConnectUrl = '')
+    {
+        if ( ! empty($giveConnectUrl)) {
+            $this->apiUrl = $giveConnectUrl;
+        } elseif (defined('GIVE_CONNECT_URL') && GIVE_CONNECT_URL) {
+            $this->apiUrl = GIVE_CONNECT_URL;
+        }
+    }
+
+    /**
+     * Get rest api endpoint url for requests.
+     *
+     * @unreleased Set default endpoint value as empty
+     * @since      2.8.0
+     *
+     * @param string $endpoint The route on the server. E.g: 'gateway-name/connect'
+     */
+    public function getApiUrl(string $endpoint = ''): string
+    {
+        if ( ! empty($endpoint)) {
+            return $this->apiUrl . '/' . ltrim($endpoint, '/');
```

Comment:
This should use [trailingslashit()](https://developer.wordpress.org/reference/functions/trailingslashit/) to ensure that a trailing slash is added and only one trailing slash is added. This protects against passing in a value that already contains a trailing slash.

Reply:
Resolved in e828c2f884f27896b1d2dee242519110a0cb164b